src/frequenz/sdk/microgrid/component/battery_utils.py

```
@@ -0,0 +1,60 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Util functions for battery."""
+
+import logging
+from typing import Dict, Optional, Set
+
+from .. import get as get_microgrid
+from . import ComponentCategory
+
+_logger = logging.getLogger(__name__)
+
+
+def find_adjacent_inverter_id(battery_id: int) -> Optional[int]:
+    """Find inverter adjacent to this battery.
+
+    Args:
+        battery_id: battery id adjacent to the wanted inverter
+
+    Note:
+        Method doesn't check if given id belongs to the battery.
+
+    Returns:
+        Id of the adjacent inverter. Return None if battery has not adjacent inverter.
+    """
+    graph = get_microgrid().component_graph
+    try:
+        inverter_id = next(
+            (
+                comp.component_id
+                for comp in graph.predecessors(battery_id)
+                if comp.category == ComponentCategory.INVERTER
+            ),
+            None,
+        )
+    except KeyError as err:
+        _logger.warning(err)
+        inverter_id = None
+
+    return inverter_id
+
+
+def battery_inverter_mapping(batteries: Set[int]) -> Dict[int, int]:
```

If we emit the log message in `find_adjacent()` this function is reduced to:

```py
bat_inv_map = {bat_id: graph.find_adjacent(bat_id, ComponentCategory.INVERTER)
    for battery_id in batteries}:
```

I'm not sure we need a function for this, unless it is something that will be used very often in many places.

If we want to keep it as a function, I think it could also be generalized as a `ComponentGraph` method, so users that want exactly what this function does could do (I would default to raise an exception if a mapping is missing, as is the safest choice):

```py
bat_inv_map = component_graph.map_adjacents(battery_ids, ComponentCategory.INVERTER,
    ignore_missing=True)
```

---

Component graph has already method to find successors and predecessors.
They are well defined and I won't add unknown `adjacent` there.
This method is highly related with battery because we are sure inverter is direct predecessor of the battery.

I moved battery_inverter_mapping to the battery_pool module. It is now private.